ESP8266/ESP8266.cpp

```diff
@@ -332,7 +334,10 @@ nsapi_error_t ESP8266::send(int id, const void *data, uint32_t amount)
         if (_parser.send("AT+CIPSEND=%d,%lu", id, amount)
             && _parser.recv(">")
             && _parser.write((char*)data, (int)amount) >= 0) {
-            while (_parser.process_oob()); // multiple sends in a row require this
+            _send_in_progress = true;
+            // wait for "SEND OK" to be received
+            // multiple sends in a row require this
+            while (_parser.process_oob() && _send_in_progress); 
```

Comment:
"If the connection cannot be established or gets
disrupted during data transmission, the system
returns:
ERROR
If data is transmitted successfully, the system
returns:
SEND OK" - https://www.espressif.com/sites/default/files/documentation/4a-esp8266_at_instruction_set_en.pdf

This loop won't terminate if a send fails. The error case should also be handled.

Reply:
I can catch the `SEND FAIL` message, but wouldn't the `ERROR` message fall under https://jira.arm.com/browse/ONME-3423?